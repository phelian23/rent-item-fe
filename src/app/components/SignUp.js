import React, { useState } from 'react';
import { Button } from 'react-bootstrap';
import { Link, useNavigate } from 'react-router-dom';
import { useDispatch } from 'react-redux';

import { signUp } from '../redux/actions/authentications';
import { validation } from './Validation';

/* eslint-disable jsx-a11y/label-has-associated-control */

const SignUp = () => {
  const [newUser, setNewUser] = useState({
    name: '',
    email: '',
    password: '',
    password_confirmation: '',
  });

  const [errors, setErrors] = useState({});

  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleChange = (e) => {
    setNewUser({
      ...newUser,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const errors = validation(newUser);
    setErrors(errors);
    const userData = {
      user: newUser,
    };

    if (Object.keys(errors).length === 0) {
      dispatch(signUp(userData));
      setTimeout(() => {
        navigate('/signin');
      }, 3000);
    }
  };

  return (
    <div className="auth-con">
      <div className="auth-form">
        <div className="auth-cont">
          <h1>Sign Up</h1>
          <form onSubmit={handleSubmit}>
            <div className="form-group m-3">
              <label htmlFor="exampleInputEmail1">Name</label>
              <input
                type="text"
                name="name"
                id="name"
                className="form-input1"
                placeholder="Enter name"
                onChange={handleChange}
                value={newUser.name}
              />
              {errors.name && <p className="text-danger">{errors.name}</p>}
            </div>
            <div className="form-group mb-3 mx-3 mx-lg-0">
              <label htmlFor="exampleInputEmail1">Email</label>
              <input
                type="email"
                name="email"
                id="email"
                className="form-input1"
                placeholder="Enter email"
                onChange={handleChange}
                value={newUser.email}
              />
              {errors.email && <p className="text-danger">{errors.email}</p>}
            </div>
            <div className="form-group mb-3 mx-3 mx-lg-0">
              <label htmlFor="password">Password</label>
              <input
                type="password"
                name="password"
                id="password"
                className="form-input1"
                placeholder="Password"
                onChange={handleChange}
                value={newUser.password}
              />
              {errors.password && (
                <p className="text-danger">{errors.password}</p>
              )}
            </div>
            <div className="form-group mb-3 mx-3 mx-lg-0">
              <label htmlFor="password2">Confirm Password</label>
              <input
                type="password"
                name="password_confirmation"
                id="password2"
                className="form-input1"
                placeholder="Confirm Password"
                onChange={handleChange}
                value={newUser.password2}
              />
              {errors.password2 && (
                <p className="text-danger">{errors.password2}</p>
              )}
            </div>
            <Button className="form-btn" type="submit">
              Sign Up
            </Button>
          </form>
          <p>Already have an account?</p>
          <Button className="form-btn" as={Link} to="/signin">
            Sign In
          </Button>
        </div>
      </div>
    </div>
  );
};

export default SignUp;
