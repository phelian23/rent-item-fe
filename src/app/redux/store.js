import { configureStore } from '@reduxjs/toolkit';
import { combineReducers, compose, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import logger from 'redux-logger';
import itemReducer from './reducers/itemReducer';
import userReducer from './reducers/userReducer';
import favouriteReducer from './reducers/favouriteReducer';

const reducer = combineReducers({
  items: itemReducer,
  user: userReducer,
  favourites: favouriteReducer,

  // Add other reducers here
});

const composed = compose(applyMiddleware(thunkMiddleware, logger));

const store = configureStore({ reducer, composed });

export default store;
