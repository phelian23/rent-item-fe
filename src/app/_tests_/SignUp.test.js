import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import SignUp from '../components/SignUp';

describe('SignUp', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <SignUp />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
