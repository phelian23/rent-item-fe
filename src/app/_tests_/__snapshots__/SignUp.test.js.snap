// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<div
  className="auth-con"
>
  <div
    className="auth-form"
  >
    <div
      className="auth-cont"
    >
      <h1>
        Sign Up
      </h1>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group m-3"
        >
          <label
            htmlFor="exampleInputEmail1"
          >
            Name
          </label>
          <input
            className="form-input1"
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Enter name"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group mb-3 mx-3 mx-lg-0"
        >
          <label
            htmlFor="exampleInputEmail1"
          >
            Email
          </label>
          <input
            className="form-input1"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group mb-3 mx-3 mx-lg-0"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="form-input1"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group mb-3 mx-3 mx-lg-0"
        >
          <label
            htmlFor="password2"
          >
            Confirm Password
          </label>
          <input
            className="form-input1"
            id="password2"
            name="password_confirmation"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
          />
        </div>
        <button
          className="form-btn btn btn-primary"
          disabled={false}
          type="submit"
        >
          Sign Up
        </button>
      </form>
      <p>
        Already have an account?
      </p>
      <a
        className="form-btn btn btn-primary"
        disabled={false}
        href="/signin"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Sign In
      </a>
    </div>
  </div>
</div>
`;
